crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2593,6 +2596,10 @@ pub enum KnownInstanceType<'db> {
     TypeGuard,
     TypeIs,
     ReadOnly,
+    KnotExtensionsUnknown,
+    KnotExtensionsNot,
+    KnotExtensionsIntersection,
+    KnotExtensionsTypeOf,
```

Comment:
I have mixed feelings even about using the `KnotExtensions` prefix here; we don't do that for special forms from other modules, unless it's necessary for other reasons (e.g. `TypingSelf`, due to Rust keyword `Self`), and I can easily see any of these becoming not red-knot-specific in future. But I don't feel strongly, it's easy to remove this prefix in future.